torch/_inductor/ir.py

```diff
@@ -2573,7 +2592,94 @@ def is_channels_last_stride_ordered(self):
         order = [len(order)] + order
         return self.is_stride_ordered(order)
 
+    @staticmethod
+    def _pad_strides(in_strides, size, align=DEFAULT_ALIGN):
+        """
+        The padding does not change stride order but makes sure all strides larger
+        than the threshold are multiple of align.
+        """
+        if len(in_strides) == 0:
+            return in_strides
+
+        # a loose check for channels last
+        if not config.pad_channels_last and len(in_strides) == 4 and in_strides[1] == 1:
+            return in_strides
+
+        current_fx_node = V.get_current_node()
+        if hasattr(current_fx_node, "meta") and current_fx_node.meta.get(
+            "dislike_padding", False
+        ):
+            return in_strides
+
+        # get_stride_order does not work with dynamic shape. Also we can not
+        # statically decide if a padding is needed or how much padding we should
+        # do for dynamic shape.
```

Comment:
Couldn't we pad to an aligned multiple even with dynamic shape ?

Reply:
We are not simply padding each stride here. We need consider stride order (check test_pad_3d_tensor for why that's the case). For dynamic shape, get_stride_order just does not work.

There may be other complexity that I have not considered yet.

Also I found we don't pad mm for dynamic shape either :)